internal/codegen/kotlin/gen.go

```diff
@@ -26,7 +28,7 @@ func sameTableName(n *ast.TableName, f core.FQN) bool {
 	if n.Schema == "" {
 		schema = "public"
 	}
-	return n.Catalog == n.Catalog && schema == f.Schema && n.Name == f.Rel
+	return n.Catalog == f.Catalog && ((f.Schema != nil && f.Schema.Match(schema)) || schema == "") && ((f.Rel != nil && f.Rel.Match(n.Name)) || n.Name == "")
```

Comment:
On the syntax question, I think we can solve this with documentation, and I'm not sure that we can reasonably solve it otherwise.

Specifically, \\\* will match a literal \* instead of acting as a wildcard. (Assuming that I did my escapes properly for the markdown.)

We could, admittedly, pick a different character with other special meaning to SQL that would generally prevent it from being a valid schema, table, or column character, but given my surprise by #957, I'd have to do some digging to see if something like \_ was valid SQL if properly escaped.

Reply:
Agreed, and with the change to overrides, we probably want a 'func (o \*Override) SameTableName (n \*ast.TableName) bool {}' to handle all override comparison.

Reply:
This has been addressed in the rework, I'll give details below.